test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -0,0 +1,302 @@
+	intakeTick              = 20 * time.Second
+	fakeintake1Name         = "1fakeintake"
+	fakeintake2Name         = "2fakeintake"
+)
+
+//go:embed testfixtures/custom_logs.yaml
+var customLogsConfig string
+
+func multiFakeintakeStackDef(agentOptions ...agentparams.Option) *e2e.StackDefinition[multiFakeIntakeEnv] {
+	return e2e.EnvFactoryStackDef(func(ctx *pulumi.Context) (*multiFakeIntakeEnv, error) {
+		awsEnv, err := awsResources.NewEnvironment(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter1, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake1Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter2, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake2Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		agentInstaller, err := agent.NewInstaller(vm, agentOptions...)
+		if err != nil {
+			return nil, err
+		}
+
+		return &multiFakeIntakeEnv{
+			VM:          client.NewPulumiStackVM(vm),
+			Agent:       client.NewPulumiStackAgent(agentInstaller),
+			Fakeintake1: client.NewFakeintake(fiExporter1),
+			Fakeintake2: client.NewFakeintake(fiExporter2),
+		}, nil
+	})
+}
+
+type multiFakeIntakeSuite struct {
+	e2e.Suite[multiFakeIntakeEnv]
+}
+
+func TestMultiFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &multiFakeIntakeSuite{}, multiFakeintakeStackDef())
+}
+
+// SetupSuite waits for both fakeintakes to be ready before running tests.
+func (v *multiFakeIntakeSuite) SetupSuite() {
+	v.Env() // update the environment outside EventuallyWithT
+
+	// Wait for the fakeintake to be ready to avoid 503
+	require.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+		assert.NoError(c, v.Env().Fakeintake1.Client.GetServerHealth())
+		assert.NoError(c, v.Env().Fakeintake2.Client.GetServerHealth())
+	}, intakeMaxWaitTime, intakeTick)
+}
+
+// BeforeTest flushes both fakeintakes before starting each test.
+func (v *multiFakeIntakeSuite) SetupTest() {
+	v.NoError(v.Env().Fakeintake1.FlushServerAndResetAggregators())
+	v.NoError(v.Env().Fakeintake2.FlushServerAndResetAggregators())
+}
+
+// TestNSSFailover tests that the agent correctly picks-up an NSS change of the intake.
+//
+// The test uses two fakeintakes to represent two backends, and the /etc/hosts file for the NSS source,
+// setting-up an entry for the intake, pointing to the first fakeintake, then changing that entry
+// to point to the second fakeintake without restarting the agent.
+//
+// The test checks that metrics, logs, and flares are properly received by the first intake before
+// the failover, and by the second one after.
+//
+// Note: although the man page of `nsswitch.conf` states that each process using it should only read
+// it once (ie. no reload), glibc and Go reload it periodically
+// cf. https://go-review.googlesource.com/c/go/+/448075
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Great having links to documentation !